virtcontainers/sandbox.go

```diff
@@ -1389,18 +1419,49 @@ func (s *Sandbox) HotplugAddDevice(device api.Device, devType config.DeviceType)
 func (s *Sandbox) HotplugRemoveDevice(device api.Device, devType config.DeviceType) error {
 	switch devType {
 	case config.DeviceVFIO:
-		vfioDevice, ok := device.(*drivers.VFIODevice)
+		vfioDevices, ok := device.GetDeviceDrive().([]*config.VFIODrive)
 		if !ok {
 			return fmt.Errorf("device type mismatch, expect device type to be %s", devType)
 		}
-		_, err := s.hypervisor.hotplugRemoveDevice(*vfioDevice, vfioDev)
-		return err
+		removedDev := []*config.VFIODrive{}
+		var err error
+		defer func() {
+			// if err happens,roll back and add the removed devices back!
+			if err != nil {
+				for _, dev := range removedDev {
+					if _, rollbackErr := s.hypervisor.hotplugAddDevice(dev, vfioDev); rollbackErr != nil {
```

Comment:
Do we care about rolling back such a thing ? I mean the real scenario here is that removing will not work for some reasons, and it will return an error. What we expect after this is a `delete` to work, and I don't think trying to re-add the device will help with that.

Reply:
I think you're right, will remove the rollback logic. Thanks!